### ISSUE TITLE
Copy csv or excel files - end of file detection issue

### ISSUE THREAD

**Issue by Sognineibicchieri** [audreyt/ethercalc#435]
Hi there, just wonder if this function is supported already? when I tried with the demo on ethercalc.org. it didn't seem to work. 

By the way, it seems there's another small error on ethercalc.org. Drop a csv file to import, goes to "working..." status forever. however, it worked fine to drop an excel file. 

Thanks a lot. Look forward to receiving your response.

**Comment by eddyparkinson** [comment]
**CSV file drop**
All worked fine when I dropped a CSV on ethercalc.org
Have you got the CSV, maybe something in the CSV is causing the error. 

**Paste**
It pastes TAB separated, not CSV. You can use the "clipboard" tab to paste CSV.

**Comment by ukigo** [comment]
Dear Eddy, 

Thanks a lot for your reply. I understood now how it works the copy and paste. 

For the csv file drop. I just tried with my test.csv file, it still didn't work. I managed to open the same the same test.csv with the curl api. I checked the csv too, it's a simple csv comma separated with only 11 lines. I saved the file with MS Excel using csv format. Then I saved the same file using .xlsx format with Excel and dropped the xlsx file to the  ethercalc.org, and it worked. I just wonder what could be wrong in the file to cause the error. I attached my file (zipped for your reference).  Thank you very much again for your time. 

Have a great day. 

[TEST.zip](https://github.com/audreyt/ethercalc/files/494956/TEST.zip)

**Comment by eddyparkinson** [comment]
Yes, there is a bug. Adding a new line to the end of the CSV solves the problem. Looks like something to do with detecting the end of the file.

**Comment by ukigo** [comment]
Thanks a lot Eddy. It didn't work either for me adding a new line. But it's great that you've labeled it as a bug. Hope that you'll be able to resolve it soon. 

Thanks again for your kind help.
